### ISSUE TITLE
0.11.0 sdist is invalid for Py3.8

### ISSUE THREAD

**Issue by anntzer** [statsmodels/statsmodels#6472]
#### Describe the bug

pip-installing statsmodels from the 0.11.0 sdist into a clean Py3.8 venv results in an error.

#### Code Sample, a copy-pastable example if possible

In a clean venv:
```
$ pip install --no-binary=:all: statsmodels==0.11.0
Collecting statsmodels
  Using cached statsmodels-0.11.0.tar.gz (15.4 MB)
  Installing build dependencies ... error
  ERROR: Command errored out with exit status 1:
   command: /tmp/testenv/bin/python /tmp/testenv/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-p1e06a75/overlay --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'cython>=0.29.14' 'numpy==1.14.5; python_version=='"'"'3.5'"'"'' 'numpy==1.14.5; python_version=='"'"'3.6'"'"'' 'numpy==1.14.5; python_version>='"'"'3.7'"'"'' 'numpy==1.17.5; python_version>='"'"'3.8'"'"'' 'scipy>=1.0'
       cwd: None
  Complete output (3 lines):
  Ignoring numpy: markers 'python_version == "3.5"' don't match your environment
  Ignoring numpy: markers 'python_version == "3.6"' don't match your environment
  ERROR: Double requirement given: numpy==1.17.5 (already in numpy==1.14.5, name='numpy')
  ----------------------------------------
ERROR: Command errored out with exit status 1: /tmp/testenv/bin/python /tmp/testenv/lib/python3.8/site-packages/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-p1e06a75/overlay --no-warn-script-location --no-binary :all: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'cython>=0.29.14' 'numpy==1.14.5; python_version=='"'"'3.5'"'"'' 'numpy==1.14.5; python_version=='"'"'3.6'"'"'' 'numpy==1.14.5; python_version>='"'"'3.7'"'"'' 'numpy==1.17.5; python_version>='"'"'3.8'"'"'' 'scipy>=1.0' Check the logs for full command output.
```

This is because pyproject.toml contains the following incompatible lines:
```
    "numpy==1.14.5; python_version>='3.7'",
    "numpy==1.17.5; python_version>='3.8'",
```

I do realize this is fixed in master, but perhaps a new sdist for 0.11 would be nice.

#### Expected Output

statsmodels installs correctly.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

N/A

**Comment by bashtage** [comment]
Not that it is great, but old pip might still work. This didn't get caught because the multibuild was happy to build the wheel and install it.

**Comment by bashtage** [comment]
I think 0.11.1 will come out next week that will fix this. Hopefully, there will be a couple of new bug fixes in as well.

**Comment by ianbeyst** [comment]
Installing with the `--no-build-isolation` flag seems to bypass the issue.

**Comment by bashtage** [comment]
Thanks for this.  I'm going to try and get 0.11.1 out this week with this fix.

**Comment by bashtage** [comment]
This should be fixed now that 0.11.1 is out.
